core/common/src/Instant.kt

```diff
@@ -25,12 +26,128 @@ import kotlin.time.*
+ * val timeUntilConcert = concertStart - start
+ * ```
+ *
+ * #### Calendar-based
+ *
+ * Since [Instant] represents a point in time, it is always well-defined what the result of arithmetic operations on it
+ * is, including the cases when a calendar is used.
+ * This is not the case for [LocalDateTime], where the result of arithmetic operations depends on the time zone.
+ * See the documentation of [LocalDateTime] for more details.
+ *
+ * Adding and subtracting calendar-based units can be done using the [plus] and [minus] operators,
+ * requiring a [TimeZone]:
+ *
+ * ```
+ * Clock.System.now().plus(1, DateTimeUnit.DAY, TimeZone.of("Europe/Berlin")) // one day from now in Berlin
+ * ```
+ *
+ * The difference between [Instant] values in terms of calendar-based units can be obtained using the [periodUntil]
+ * method:
+ *
+ * ```
+ * val start = Clock.System.now()
+ * val concertStart = LocalDateTime(2023, 1, 1, 20, 0, 0).toInstant(TimeZone.of("Europe/Berlin"))
+ * val timeUntilConcert = start.periodUntil(concertStart, TimeZone.of("Europe/Berlin"))
+ * // two months, three days, four hours, and five minutes until the concert
+ * ```
+ *
+ * or [Instant.until] method, as well as [Instant.daysUntil], [Instant.monthsUntil],
+ * and [Instant.yearsUntil] extension functions:
+ *
+ * ```
+ * val start = Clock.System.now()
+ * val concertStart = LocalDateTime(2023, 1, 1, 20, 0, 0).toInstant(TimeZone.of("Europe/Berlin"))
+ * val timeUntilConcert = start.until(concertStart, DateTimeUnit.DAY, TimeZone.of("Europe/Berlin"))
+ * // 63 days until the concert, rounded down
+ * ```
+ *
+ * ### Construction, serialization, and deserialization
+ *
+ * [fromEpochSeconds] can be used to construct an instant from the number of seconds since
+ * `1970-01-01T00:00:00Z` (the Unix epoch).
+ * [epochSeconds] and [nanosecondsOfSecond] can be used to obtain the number of seconds and nanoseconds since the epoch.
+ *
+ * [fromEpochMilliseconds] allows constructing an instant from the number of milliseconds since the epoch.
+ * [toEpochMilliseconds] can be used to obtain the number of milliseconds since the epoch.
+ * Note that [Instant] supports nanosecond precision, so converting to milliseconds is a lossy operation.
+ *
+ * [parse] and [toString] methods can be used to obtain an [Instant] from and convert it to a string in the
+ * ISO 8601 extended format (for example, `2023-01-02T22:35:01+01:00`).
+ * During parsing, the UTC offset is not returned separately.
+ * If the UTC offset is important, use [DateTimeComponents] with [DateTimeComponents.Formats.ISO_DATE_TIME_OFFSET] to
+ * parse the string instead.
+ *
+ * Additionally, there are several `kotlinx-serialization` serializers for [Instant]:
+ * - [InstantIso8601Serializer] for the ISO 8601 extended format,
+ * - [InstantComponentSerializer] for an object with components.
+ *
+ * @see LocalDateTime for a user-visible representation of moments in time in an unspecified time zone.
```

Comment:
This would look nice in the platform-specific documentation: then, we'd also mention `NSDate`, and maybe some JS thing, etc., along with the converters. In the common documentation, I'm not so sure.